feat: touch up callout component

Creeland opened the pull request:
Made a few tweaks to the Callout component:

Added text prop for easier text rendering, falls back to children if not provided. I noticed that the callout had some whitespace due to "children" not just being the text, but the html that was created from the text when the markdown was compiled. 

I also cleaned up the component and added some additional text styling. 

## Old

<img width="2560" alt="image" src="https://user-images.githubusercontent.com/518406/232126599-5a96d807-5656-40a0-8fa8-503b0cb064b7.png">
<img width="2560" alt="image" src="https://user-images.githubusercontent.com/518406/232126659-3f8ebac6-8784-4a88-aaea-cc91db022769.png">


## New

<img width="2560" alt="image" src="https://user-images.githubusercontent.com/518406/232126712-8baedceb-723d-48fb-a5e6-3f006d1c8a90.png">
<img width="2560" alt="image" src="https://user-images.githubusercontent.com/518406/232126741-e05b8334-9867-4234-ba1e-9f2db4020f13.png">

vojtaholik commented on src/components/mdx/callout.js:
```diff
@@ -1,13 +1,9 @@
   return (
     <div
-      className={`mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 overflow-hidden border-blue-900 dark:text-blue-200 text-blue-600 sm:px-8 px-4 py-3 rounded-sm`}
+      className={`leading-relaxed mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 border-blue-900  sm:px-8 px-4 py-6 rounded-sm font-small sm:text-lg text-base tracking-normal shadow-md ${className}`}
```
`font-small` is not a valid tailwind class

Creeland commented:
Updated the component to be more consistent with our horizontal resource card spacing. I also dropped the class that didn't exist that I had mistakenly used @vojtaholik 

![image](https://user-images.githubusercontent.com/518406/232133069-d5243295-33fa-4887-ae1e-56ceffc38abf.png)
  
![image](https://user-images.githubusercontent.com/518406/232133086-7605212f-82a6-4497-b3cc-e6e7130f006c.png)

vojtaholik commented on src/components/mdx/callout.js:
```diff
@@ -1,13 +1,9 @@
   return (
     <div
-      className={`mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 overflow-hidden border-blue-900 dark:text-blue-200 text-blue-600 sm:px-8 px-4 py-3 rounded-sm`}
+      className={`p-5 leading-relaxed mx-auto my-3 w-full dark:bg-gray-800 bg-blue-100 border-l-4 border-blue-900 sm:space-x-5 space-x-0 sm:space-y-0 space-y-5 items-center text-left shadow-sm rounded-lg overflow-hidden lg:text-lg sm:text-md ${className}`}
```
using `className` like this without making it an empty string as a fallback will make it return `undefined` by default.

vojtaholik commented on src/components/mdx/callout.js:
```diff
@@ -1,13 +1,9 @@
-const container = ''
-
-export default function Callout({children, className}) {
+export default function Callout({children, text, className}) {
```
why not convert it to TS while you're there? :)

vojtaholik commented:
I used actual links, fixed horizontal scrolling caused by code blocks by reducing their negative horizontal margin, and made the callouts stretch to full width on mobile.

<img src="https://user-images.githubusercontent.com/25487857/234482431-492b45b0-de2d-482a-95c6-5217693f8dce.png" width="450" alt="screenshots">

vojtaholik commented on src/components/mdx/callout.tsx:
```diff
@@ -0,0 +1,29 @@
+        )}
+        aria-label="callout"
+      >
+        {text ? <p>{text}</p> : children}
```
we want paragraphs for seo and a11y
